pkg/pgmodel/migrations/sql/idempotent/base.sql

```diff
@@ -430,6 +430,29 @@ CREATE OR REPLACE FUNCTION SCHEMA_CATALOG.label_unnest(label_array anyarray)
 AS $function$ SELECT unnest(label_array) $function$;
 GRANT EXECUTE ON FUNCTION SCHEMA_CATALOG.label_unnest(anyarray) to prom_reader;
 
+-- approximate_row_count is used to count the approximate number of rows in a table (non-hypertable).
+CREATE OR REPLACE FUNCTION SCHEMA_PROM.approximate_row_count(table_name REGCLASS)
+    RETURNS BIGINT
+    LANGUAGE SQL
+AS
+$$
+    SELECT reltuples::BIGINT FROM pg_class WHERE oid=table_name;
+$$ STABLE PARALLEL SAFE;
+
+-- safe_approximate_row_count returns the approximate row count of a hypertable if timescaledb is installed
+-- else returns the approximate row count in the normal table. This prevents errors in approximate count calculation
+-- if timescaledb is not installed, which is the case in plain postgres support.
+CREATE OR REPLACE FUNCTION SCHEMA_CATALOG.safe_approximate_row_count(table_name REGCLASS) RETURNS BIGINT
+    LANGUAGE PLPGSQL
+AS
+$$
+BEGIN
+    IF SCHEMA_CATALOG.is_timescaledb_installed() THEN
+        RETURN (SELECT row_estimate FROM hypertable_approximate_row_count(table_name));
+    END IF;
+    RETURN (SELECT SCHEMA_PROM.approximate_row_count(table_name));
```

Comment:
I don't understand why we have a separate function here instead of just writing the sql in this function?

Reply:
Its like to have a function that is smart enough to know that when demanded for approx row cunt, then based on the situation (timescaledb or vanilla postgres), decide and give the rows. this reduces the callers length and makes thing concise.

As per discussion, we should check if its a hypertable than checking if timescaledb installed.